community/kernel/src/main/java/org/neo4j/kernel/impl/locking/ResourceTypes.java

```diff
@@ -135,4 +159,88 @@ public static ResourceType fromId( int typeId )
     {
         return idToType.get( typeId );
     }
+
+    /**
+     * This is a stronger, full 64-bit hashing method for schema index entries that we should use by default in a
+     * future release, where we will also upgrade the HA protocol version. Currently this is indicated by the "4_x"
+     * name suffix, but any version where the HA protocol version changes anyway would be just as good an opportunity.
+     *
+     * @see IncrementalXXH64
+     */
+    public static long indexEntryResourceId_4_x( long labelId, IndexQuery.ExactPredicate... predicates )
+    {
+        long hash = IncrementalXXH64.init( 0x0123456789abcdefL );
```

Comment:
We already have HashFunction defined that is used in a HopScotch, i would say that IncrementalXXH64 should be changed to be out of the possible HashFunctions and they should be moved to commons module most probably, and that interface can be extended to include `update(hash, block)` and `finalise(hash)` so new hashing function will not gonna be glued that much to ResourceTypes